### PR TITLE
chore(labeler): upgrade to version 5 syntax

### PR DESCRIPTION
**Pull request by dndx**
https://github.com/actions/labeler?tab=readme-ov-file#breaking-changes-in-v5

This should address the PR workflow failure due to workflow bump. e.g. https://github.com/Kong/kong/actions/runs/7204838757/job/19627484166?pr=12209

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -1,190 +1,231 @@
...
 
 core/db/migrations:
-- kong/db/migrations/**/*
+- changed-files:
+  - any-glob-to-any-file: kong/db/migrations/**/*
 
 core/db:
-- any: ['kong/db/**/*', '!kong/db/migrations/**/*']
+- changed-files:
+  - any-glob-to-any-file: ['kong/db/**/*', '!kong/db/migrations/**/*']
```
Comment (AndyZhang0707) [inline .github/labeler.yml:27]:
```suggestion
  - any-glob-to-any-file: 'kong/db/**/*'
  - all-globs-to-all-files: '!kong/db/migrations/**/*'
```

File: .github/labeler.yml [hunk 2]
```diff
@@ -1,190 +1,231 @@
...
 
 changelog:
-- CHANGELOG.md
+- changed-files:
+  - any-glob-to-any-file: CHANGELOG.md
 
 core/docs:
-- any: ['**/*.md', '!CHANGELOG.md']
+- changed-files:
+  - any-glob-to-any-file:['**/*.md', '!CHANGELOG.md']
```
Comment (AndyZhang0707) [inline .github/labeler.yml:35]:
```suggestion
  - any-glob-to-any-file:'**/*.md'
  - all-globs-to-all-files: '!CHANGELOG.md'
```

File: .github/labeler.yml [hunk 3]
```diff
@@ -1,190 +1,231 @@
...
 
 core/logs:
-- kong/pdk/log.lua
+- changed-files:
+  - any-glob-to-any-file: kong/pdk/log.lua
 
 core/pdk:
-- any: ['kong/pdk/**/*', '!kong/pdk/log.lua']
+- changed-files:
+  - any-glob-to-any-file: ['kong/pdk/**/*', '!kong/pdk/log.lua']
```
Comment (AndyZhang0707) [inline .github/labeler.yml:59]:
```suggestion
  - any-glob-to-any-file: 'kong/pdk/**/*'
  - all-globs-to-all-files: '!kong/pdk/log.lua'
```

File: .github/labeler.yml [hunk 4]
```diff
@@ -1,190 +1,231 @@
...
 
 core/proxy:
-- any: ['kong/runloop/**/*', '!kong/runloop/balancer/*', '!kong/runloop/plugin_servers/*']
+- changed-files:
+  - any-glob-to-any-file: ['kong/runloop/**/*', '!kong/runloop/balancer/*', '!kong/runloop/plugin_servers/*']
```
Comment (AndyZhang0707) [inline .github/labeler.yml:63]:
```suggestion
  - any-glob-to-any-file: 'kong/runloop/**/*'
  - all-globs-to-all-files: '!kong/runloop/balancer/*'
  - all-globs-to-all-files: '!kong/runloop/plugin_servers/*'
```

**Review by AndyZhang0707 — CHANGES_REQUESTED** [review]

**Review by outsinre — APPROVED** [review]

**Comment by dndx** [comment]
Effect: https://github.com/dndx/labeler-test/pull/1

**Review by AndyZhang0707 — APPROVED** [review]
LGTM

**Pull request merged**